packages/frontend/component/src/ui/modal/modal.tsx

```diff
@@ -48,141 +45,178 @@ const getVar = (style: number | string = '', defaultValue = '') => {
     : defaultValue;
 };
 
-export const Modal = forwardRef<HTMLDivElement, ModalProps>(
-  (
-    {
-      width,
-      height,
-      minHeight = 194,
-      title,
-      description,
-      withoutCloseButton = false,
-      modal,
-      persistent,
-      animationTimeout = 120,
-      portalOptions,
-      open: customOpen,
-      onOpenChange: customOnOpenChange,
-      contentOptions: {
-        style: contentStyle,
-        className: contentClassName,
-        onPointerDownOutside,
-        onEscapeKeyDown,
-        ...otherContentOptions
-      } = {},
-      overlayOptions: {
-        className: overlayClassName,
-        style: overlayStyle,
-        ...otherOverlayOptions
-      } = {},
-      closeButtonOptions = {},
-      children,
-      ...props
-    },
-    ref
-  ) => {
-    const [{ status }, toggle] = useTransition({
-      timeout: animationTimeout,
-      onStateChange: useCallback(
-        ({ current }: { current: TransitionState }) => {
-          // add more status if needed
-          if (current.status === 'exited') customOnOpenChange?.(false);
-          if (current.status === 'entered') customOnOpenChange?.(true);
-        },
-        [customOnOpenChange]
-      ),
+/**
+ * This component is a hack to support `startViewTransition` in the modal.
+ */
+class ModalTransitionContainer extends HTMLElement {
```

Comment:
I think we can use Lit here

Reply:
I think this is enough since we don't need to render anything. Are there any benefits to using lit?

Reply:
maybe not. I think lit's syntax may make the code more readable when defining wc.